### PR TITLE
Product

### PR DESCRIPTION
**Pull request by Piyushg094**

### CODE REVIEW

File: admin-ui/src/app/pages/Category/Category.js [hunk 1]
```diff
@@ -26,6 +27,9 @@ const CategoryList = () => {
   const [dialogOpen, setDialogOpen] = useState(false);
   const [bulkaction, setbulkaction] = useState(null)
 
+  console.log("piyush")
```
Comment (subhamsahu) [inline admin-ui/src/app/pages/Category/Category.js:30]: Remove these kinds of unnecessary logging

**Comment by Piyushg094** [comment]
Ohky sir

File: admin-ui/src/app/pages/Category/components/CategoryCreateForm.js [hunk 2]
```diff
@@ -62,22 +62,46 @@ const CategoryCreateForm = ({ setloading, fetchAllCategories }) => {
         setloading(false);
     }
 
+    // const handleImageUpload = async () => {
+    //     if (selectedFiles && selectedFiles.length>0) {
```
Comment (subhamsahu) [inline admin-ui/src/app/pages/Category/components/CategoryCreateForm.js:66]: Remove Unnecessary Function

File: admin-ui/src/app/pages/Product/components/CreateForm.js [hunk 3]
```diff
@@ -26,36 +32,97 @@ const createPayload = {
...
+          
+      } 
+    }
 
-    const handleChange = (event) => {
-        setAge(event.target.value);
-    };
+    const categories = useSelector((state)=>state.categoryListDataReducer.categoryList.data.categories);
+    
+
```
Comment (subhamsahu) [inline admin-ui/src/app/pages/Product/components/CreateForm.js:72]: Remove

File: admin-ui/src/app/pages/Product/components/CreateForm.js [hunk 4]
```diff
@@ -26,36 +32,97 @@ const createPayload = {
...
+        if (e.target.name === 'isActive' || e.target.name === 'taxable'  ) {
             setproductPayload({ ...productPayload, [e.target.name]: e.target.checked })
         }
         else {
             setproductPayload({ ...productPayload, [e.target.name]: e.target.value })
         }
+        if(e.target.name==='category')
+        {
+          setproductPayload({...productPayload , [e.target.name]: e.target.value})
+          setSelectedCategory(e.target.value)
```
Comment (subhamsahu) [inline admin-ui/src/app/pages/Product/components/CreateForm.js:96]: 2 Condtions , use elif

File: admin-ui/src/app/pages/Product/components/CreateForm.js [hunk 5]
```diff
@@ -26,36 +32,97 @@ const createPayload = {
...
     const handleCreateProduct = async () => {
-        setloading(true);
+        setloading(true); 
+        
         const image_data = await handleImageUpload()
         let payload = productPayload
+        console.log(selectedCategoryData)
+        if (selectedCategoryData)
+        {
+            console.log(selectedCategoryData)
```
Comment (subhamsahu) [inline admin-ui/src/app/pages/Product/components/CreateForm.js:123]: Unnecssary code

File: admin-ui/src/app/pages/Product/components/CreateForm.js [hunk 6]
```diff
@@ -150,25 +226,66 @@ const CreateForm = ({ setloading, fetchAllProducts }) => {
                 <Grid item xs={12}>
                     <input type="file" className='upload-list-inline' onChange={handleFileChange} />
                 </Grid>
-                <Grid item xs={4}>
+                {/* <Grid item xs={4}>
```
Comment (subhamsahu) [inline admin-ui/src/app/pages/Product/components/CreateForm.js:229]: If possible , remove unwanted code

File: admin-ui/src/app/services/Common.js [hunk 7]
```diff
@@ -92,12 +92,23 @@ export const put = async (url, payload) => {
     return data
 }
 
-export const uploadImage = async (url, formData) => {
-    const headersData = generateGlobalHeader()
-    const response = await axios.post(url, formData, headersData)
-        .then(res => { return res })
-        .catch(err => { return err });
-    const data = response?.data || []
-    return data
```
Comment (subhamsahu) [inline admin-ui/src/app/services/Common.js:101]: Check if unwanted comment

File: server/app/controllers/category.js [hunk 8]
```diff
@@ -137,9 +138,9 @@ const list = async (req, res) => {
     }
 }
 
-const listActive = async (req, res) => {
+const count = async (req, res) => {
```
Comment (subhamsahu) [inline server/app/controllers/category.js:141]: chnage count to list

File: server/app/controllers/product.js [hunk 9]
```diff
@@ -43,6 +49,12 @@ const create = async (req, res) => {
             );
         }
 
+        // if(!category){
```
Comment (subhamsahu) [inline server/app/controllers/product.js:52]: Revert and show the error

File: server/app/public/images/images-1706187664424-995346249.jpeg
Comment (subhamsahu) [inline server/app/public/images/images-1706187664424-995346249.jpeg:1]: Put some generic image

File: server/app/routes/api/category.js [hunk 10]
```diff
@@ -15,7 +15,11 @@ router.post('/add', auth(), role.checkRole(ROLES.Admin), (req, res) => {
 });
 
 // fetch store categories api
-router.get('/list/active', async (req, res) => {
+router.get('/count', async (req, res) => {
```
Comment (subhamsahu) [inline server/app/routes/api/category.js:18]: change from count to list

File: server/app/routes/api/category.js [hunk 11]
```diff
@@ -45,6 +49,12 @@ router.delete('/:id', auth(), role.checkRole(ROLES.Admin), checkValidMongoDbId()
   categoryController.remove(req, res);
 });
 
+// count categories
```
Comment (subhamsahu) [inline server/app/routes/api/category.js:52]: remove if unnecessary

File: server/app/routes/api/razorpay.js [hunk 12]
```diff
@@ -5,7 +5,7 @@ const router = express.Router();
 const auth = require('../../middleware/auth');
 const role = require('../../middleware/role');
 const { ROLES } = require('../../constants');
-const razorpayController = require('../../controllers/razorpay');
+// ?const razorpayController = require('../../controllers/razorpay')
```
Comment (subhamsahu) [inline server/app/routes/api/razorpay.js:8]: Revert and show me error

**Review by subhamsahu — CHANGES_REQUESTED** [review]
Apply all changes

**Review by subhamsahu — APPROVED** [review]

**Pull request merged**